### PR TITLE
feat: Add tests for Virtual Cards service

### PR DESCRIPTION
**Pull request by bistaastha**
## Clickup
https://app.clickup.com/t/86cuaht42

## Code Coverage
Screenshot 2024-01-25 at 3 38 31 PM

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

**Review by suyashpatil78 — APPROVED** [review]

**Comment by mvaishnavi** [comment]
my review not needed. please merge in once confirming with mobile app team.

File: src/app/core/mock-data/virtual-card-details-response.data.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import { PlatformApiResponse } from '../models/platform/platform-api-response.model';
+
+export const virtualCardDetailsResponse: PlatformApiResponse<CardDetailsResponse> = {
+  count: 1,
+  offset: 0,
+  data: [
+    {
+      full_card_number: '123412341234123',
+      cvv: '123',
+      expiry_date: new Date('2024-03-03'),
```
Comment (Julias0) [inline src/app/core/mock-data/virtual-card-details-response.data.ts:11]: You wont get date objects in the response right? You will get strings. You should be passing them to a date object inside the service method right

**Review by Julias0 — CHANGES_REQUESTED** [review]
1 comment

File: src/app/core/services/virtual-cards.service.spec.ts [hunk 2]
```diff
@@ -1,16 +1,72 @@
 import { TestBed } from '@angular/core/testing';
```
Comment (bistaastha) [inline src/app/core/services/virtual-cards.service.spec.ts:1]: Screenshot 2024-02-01 at 12 41 24 PM

Comment (bistaastha) [reply src/app/core/services/virtual-cards.service.spec.ts:1]: The unit tests failing currently are due to virtual cards component, which will be fixed in follow up

File: src/app/core/mock-data/virtual-card-details-response.data.ts (hunk 1 above)
Comment (bistaastha) [reply src/app/core/mock-data/virtual-card-details-response.data.ts]: Added the conversion in the service

File: src/app/core/mock-data/virtual-card-details-response.data.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { CardDetailsAmountResponse } from '../models/card-details-amount-response';
+import { CardDetailsResponse } from '../models/card-details-response.model';
+
+export const virtualCardDetailsResponse: { data: CardDetailsResponse } = {
+  data: {
+    full_card_number: '123412341234123',
+    cvv: '123',
+    // @ts-ignore
+    expiry_date: '2029-01-01T00:00:00+00:00',
```
Comment (bistaastha) [inline src/app/core/mock-data/virtual-card-details-response.data.ts:9]: This will be converted to Date in the service

**Review by Julias0 — APPROVED** [review]

**Pull request merged**